hq/app/logic/SnykDisplay.scala

```diff
@@ -2,63 +2,60 @@ package logic
     Json.parse(s).asOpt[SnykError].getOrElse(SnykError(s))
   } catch {
-    case _: JsonParseException => SnykError(s)
+    case e: JsonParseException => SnykError(e.getMessage)
   }
 
-  private def parseJsonToOrganisationList(s: String): Option[List[SnykOrganisation]] = try {
-    (Json.parse(s) \ "orgs").asOpt[List[SnykOrganisation]]
+  def parseJsonToObject[A](label: String, s: String, f: (String => A)): Attempt[A] = try {
+    Attempt.Right(f(s))
   } catch {
-    case _: JsonParseException => None
+    case e: JsonParseException =>
+      val f = Failure(s"Unable to find $label from $s", s"Could not read Snyk response ($s)", 502, None, Some(e))
+      Attempt.Left(f)
+    case e: Exception =>
+      val error = parseJsonToError(s)
+      val f = Failure(s"Unable to find $label from $s", s"Could not read Snyk response (${error.error})", 502, None, Some(e))
+      Attempt.Left(f)
   }
 
-  def getProjectIdList(s: String): Attempt[List[SnykProject]] = {
-    val projects = parseJsonToProjectIdList(s)
-    val error = parseJsonToError(s)
-    val f = Failure(s"Unable to find project ids from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-    Attempt.fromOption(projects, FailedAttempt(f))
-  }
+  def parseJsonToOrganisationList(s: String): Attempt[List[SnykOrganisation]] =
+    parseJsonToObject("organisations", s, body => {(Json.parse(body) \ "orgs").as[List[SnykOrganisation]]} )
 
-  private def parseJsonToProjectIdList(s: String) = try {
-    (Json.parse(s) \ "projects").asOpt[List[SnykProject]]
-  } catch {
-    case _: JsonParseException => None
+  def getProjectIdList(ss: List[(SnykOrganisation, String)])(implicit ec: ExecutionContext): Attempt[List[((SnykOrganisation, String), List[SnykProject])]] = {
+    Attempt.labelledTraverse(ss) { s =>
+      parseJsonToProjectIdList(s._2)
+    }
   }
 
-  def parseProjectVulnerabilities(projects: List[String])(implicit ec:ExecutionContext): Attempt[List[SnykProjectIssues]] = {
-    val projectVulnerabilitiesList = projects.map( s => {
-      val projectVulnerabilities = parseJsonToProjectVulnerabilities(s)
-      val error = parseJsonToError(s)
-      val f = Failure(s"Unable to find project vulnerabilities from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-      Attempt.fromOption(projectVulnerabilities, FailedAttempt(f))
-    })
-    Attempt.traverse(projectVulnerabilitiesList)(a => a)
-  }
+  def parseJsonToProjectIdList(s: String): Attempt[List[SnykProject]] =
+    parseJsonToObject("project ids", s, body => (Json.parse(body) \ "projects").as[List[SnykProject]] )
 
-  private def parseJsonToProjectVulnerabilities(s: String) = try {
-    Json.parse(s).asOpt[SnykProjectIssues]
-  } catch {
-    case _: JsonParseException => None
+  def parseProjectVulnerabilitiesBody(s: String): Attempt[SnykProjectIssues] =
+  parseJsonToObject("project vulnerabilities", s, body => Json.parse(body).as[SnykProjectIssues])
```

Comment:
Indentation is off here (and also, this parse `f` thing with `as` is probably not what we're after see above)